includes/class-wc-gateway-paystack.php

```diff
@@ -939,6 +940,7 @@ public function process_redirect_payment_option( $order_id ) {
 
 		$paystack_params['metadata']['custom_fields'] = $this->get_custom_fields( $order_id );
 		$paystack_params['metadata']['cancel_action'] = wc_get_cart_url();
+		$paystack_params['metadata']['referrer'] = get_permalink(wc_get_page_id('shop'));
```

Comment:
Instead of the shop URL, can you switch it to the site url instead `esc_url( site_url() )`